lib/twirp/protoc_plugin/code_generator.rb

```diff
@@ -52,14 +53,7 @@ def generate
           output << "\n" if index > 0
 
           service_name = service.name
-          # The generated service class name should end in "Service"; Only append the
-          # suffix if the service is not already well-named.
-          service_class_name = if service_name.end_with?("Service")
-            service_name
-          else
-            service_name + "Service"
-          end
-          service_class_name = service_class_name.camel_case
+          service_class_name = service.service_class_name
```

Comment:
does setting these variables here make it more readable? If you're only using them once, it might be even clearer inline.

Reply:
ah, you are using `service_class_name` twice at least.

Reply:
The alternative would be this, which I didn't like:

```ruby
if name.end_with?("Service")
  name
else
  name + "Service"
end.camel_case
```

IMO, that last `.camel_case` on the end gets lost.

Open to additional alternatives.

Reply:
Ooh, you're talking about another section of code.

Yeah, I captured `service_class_name` there because it's used both for the service generation and the client generation, so we save it to avoid computing it twice. It's not really a big deal either way though.